be/src/exec/hdfs_scanner_text.cpp

```diff
@@ -144,27 +156,82 @@ void HdfsScannerCSVReader::_trim_row_delimeter(Record* record) {
     }
 }
 
+char* HdfsScannerCSVReader::_find_line_delimiter(starrocks::CSVBuffer& buffer, size_t pos) {
+    // If we can get explicit line.delim from hms, we don't need to probe it
+    if (LIKELY(!_need_probe_line_delimiter)) {
+        return buffer.find(_parse_options.row_delimiter, pos);
+    } else {
+        // If we didn't get explicit line.delim from hms, we need to probe it.
+        // We will probe '\n' first, most of TextFile's line.delim is '\n'
+        // Then we will try to probe '\r'
+        // TODO(Smith)
+        // We didn't support to treat '\r\n' as line.delim,
+        // because our code does not support line separator's length larger than one char.
+        char* p = buffer.find(DEFAULT_LINE_DELIM, pos);
```

Comment:
Will we consume a \r if there is \r\n as delimiter?

Reply:
We only support single char delimiter now.